File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/pod/definition/PodStatus.java
package info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.definition;

public enum PodStatus {
    UNINITIALIZED((byte) 0x00),
    MFG_TEST((byte) 0x01),
    FILLED((byte) 0x02),
    UID_SET((byte) 0x03),
    ENGAGING_CLUTCH_DRIVE((byte) 0x04),
    CLUTCH_DRIVE_ENGAGED((byte) 0x05),
    BASAL_PROGRAM_RUNNING((byte) 0x06),
    PRIMING((byte) 0x07),
    RUNNING_ABOVE_MIN_VOLUME((byte) 0x08),
    RUNNING_BELOW_MIN_VOLUME((byte) 0x09),
    UNUSED_10((byte) 0x0a),
    UNUSED_11((byte) 0x0b),
    UNUSED_12((byte) 0x0c),
    ALARM((byte) 0x0d),
    LUMP_OF_COAL((byte) 0x0e),
    DEACTIVATED((byte) 0x0f),
    UNKNOWN((byte) 0xff);

    private byte value;

    PodStatus(byte value) {
        this.value = value;
    }

    public static PodStatus byValue(byte value) {
        for (PodStatus status : values()) {
            if (status.value == value) {
                return status;
            }
        }
        return UNKNOWN;
    }
}

File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/pod/definition/PodStatus.kt
package info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.definition

enum class PodStatus(private val value: Byte) {
    UNINITIALIZED(0x00.toByte()), MFG_TEST(0x01.toByte()), FILLED(0x02.toByte()), UID_SET(0x03.toByte()), ENGAGING_CLUTCH_DRIVE(0x04.toByte()), CLUTCH_DRIVE_ENGAGED(0x05.toByte()), BASAL_PROGRAM_RUNNING(0x06.toByte()), PRIMING(0x07.toByte()), RUNNING_ABOVE_MIN_VOLUME(0x08.toByte()), RUNNING_BELOW_MIN_VOLUME(0x09.toByte()), UNUSED_10(0x0a.toByte()), UNUSED_11(0x0b.toByte()), UNUSED_12(0x0c.toByte()), ALARM(0x0d.toByte()), LUMP_OF_COAL(0x0e.toByte()), DEACTIVATED(0x0f.toByte()), UNKNOWN(0xff.toByte());

    companion object {

        fun byValue(value: Byte): PodStatus {
            for (status in values()) {
                if (status.value == value) {
                    return status
                }
            }
            return UNKNOWN
        }
    }
}